src/k8s/pkg/k8s/client/client.go

```diff
@@ -5,64 +5,36 @@ import (
 	"fmt"
 
 	"github.com/canonical/k8s/pkg/snap"
-	"github.com/canonical/lxd/shared/api"
 	"github.com/canonical/microcluster/client"
 	"github.com/canonical/microcluster/microcluster"
 )
 
-// ClusterOpts contains options for cluster queries.
-type ClusterOpts struct {
-	// Verbose enables info level logging.
-	Verbose bool
-	// Debug enables trace level logging.
-	Debug bool
-	// Snap is the snap instance.
-	Snap snap.Snap
-	// StateDir is the directory that contains the cluster state (for local clients).
-	StateDir string
-}
-
-// k8sdClient interacts with the k8s REST-API via unix-socket or HTTPS
+// k8sdClient interacts with the k8sd REST-API via unix-socket.
 type k8sdClient struct {
 	m    *microcluster.MicroCluster
 	mc   *client.Client
 	snap snap.Snap
 }
 
-// NewClient returns a client to interact with the k8s REST-API
-// On a cluster node it will return a client connected to the unix-socket
-// elsewhere it returns a HTTPS client that expects the certificates to be located at ClusterOpts.StateDir
-func NewClient(ctx context.Context, opts ClusterOpts) (*k8sdClient, error) {
-	if opts.Snap == nil {
-		panic("opts.Snap not specified in NewClient()")
-	}
-	if opts.StateDir == "" {
-		opts.StateDir = opts.Snap.K8sdStateDir()
+// New returns a client to interact with the k8sd REST-API.
+func New(ctx context.Context, snap snap.Snap) (*k8sdClient, error) {
+	if snap == nil {
+		panic("snap must not be nil")
 	}
 	m, err := microcluster.App(ctx, microcluster.Args{
-		Debug:    opts.Debug,
-		StateDir: opts.StateDir,
-		Verbose:  opts.Verbose,
+		StateDir: snap.K8sdStateDir(),
 	})
 	if err != nil {
-		return nil, fmt.Errorf("cannot read cluster info: %w", err)
+		return nil, fmt.Errorf("failed to initialize microcluster app: %w", err)
 	}
-
-	microClient, err := m.LocalClient()
+	mc, err := m.LocalClient()
 	if err != nil {
-		return nil, fmt.Errorf("cannot create local client: %w", err)
+		return nil, fmt.Errorf("failed to create local microcluster client: %w", err)
 	}
 
 	return &k8sdClient{
-		snap: opts.Snap,
+		snap: snap,
 		m:    m,
-		mc:   microClient,
+		mc:   mc,
 	}, nil
 }
-
-func (c *k8sdClient) Query(ctx context.Context, method string, path *api.URL, in any, out any) error {
```

Comment:
nit: you could keep the wrapper (maybe make it private) and assemble to proper error message here instead of doing that manually in each call. This may become outdated.

```
func (c *k8sdClient) Query(ctx context.Context, method string, path *api.URL, in any, out any) error {
	if err := c.mc.Query(ctx, method, path, in, out); err != nil {
		return fmt.Errorf("failed to %s %s: %w, method, path, err)
	}
	return nil
}
```

Reply:
i'd rather we keep this on a client side. functions should not wrap the error with info the caller already has, and we might want to add more info in the future